mlir/include/mlir/IR/BuiltinLocationAttributes.td

```diff
@@ -18,7 +18,8 @@ include "mlir/IR/BuiltinDialect.td"
 
 // Base class for Builtin dialect location attributes.
 class Builtin_LocationAttr<string name, list<Trait> traits = []>
-    : AttrDef<Builtin_Dialect, name, traits, "::mlir::LocationAttr"> {
+    : AttrDef<Builtin_Dialect, name, traits # [NativeAttrTrait<"IsLocation">],
+              "::mlir::LocationAttr"> {
```

Comment:
Can you add a LocationAttrDef near AttrDef that adds the trait and also sets the base class to `::mlir::LocationAttr`? (Pretty much every location will want that)